src/base64_ops.rs

```diff
@@ -0,0 +1,96 @@
+use crate::allocator::{Allocator, Atom, NodePtr};
+use crate::cost::check_cost;
+use crate::cost::Cost;
+use crate::err_utils::err;
+use crate::op_utils::{atom_len, get_args, MALLOC_COST_PER_BYTE};
+use crate::reduction::{EvalErr, Reduction, Response};
+use crate::SExp;
+use base64ct::{Base64UrlUnpadded, Encoder, Encoding};
+
+const BASE64_ENCODE_BASE_COST: Cost = 40;
+const BASE64_COST_PER_ARG: Cost = 130;
+const BASE64_DECODE_BASE_COST: Cost = 400;
+const BASE64_COST_PER_BYTE: Cost = 3;
+
+const fn encoded_len(n: usize) -> Option<usize> {
```

Comment:
What is this?

Reply:
it computes the number of characters needed to base64 encode a buffer (without padding)